shell/server/proxy.js

```diff
@@ -781,6 +783,23 @@ Proxy.prototype._callNewSession = function (request, viewInfo) {
         : [ "en-US", "en" ]
   });
 
+  return this.uiView.newSession(userInfo, makeHackSessionContext(this.grainId),
+    WebSession.typeId, params).session;
+};
+
+Proxy.prototype._callNewApiSession = function (request, userInfo) {
+  var self = this;
+
+  // TODO(someday): We are currently falling back to WebSession if we get any kind of error upon
+  // calling newSession with an ApiSession._id.
+  // Eventually we'll remove this logic once we're sure apps have updated.
+  return this.uiView.newSession(userInfo, makeHackSessionContext(this.grainId),
+    ApiSession.typeId).then(function (session) {
```

Comment:
Some formatting nitpicks (which I wouldn't have commented on if the PR were ready to merge, but since it's not...):
- Continuation lines (not part of a braced block) should be indented at least four spaces, not two.
- When breaking a line in a parenthesized list, I find it easiest to read if either the continuation is indented to match the open paren, like:

        someLongFunctionName(param1, param2,
                             param3, param4);

  Or the break happens immediately after the open paren and all the parameters are indented four spaces:

      someLongFunctionName(
          param1, param2, param3, param4);

- With promises, there's a lot of possible styles and none of them are great, but I've settled on indenting the closure/lambda body as a regular block -- so, the contents are indented two spaces beyond the surrounding block. Also, if `.then(...{` doesn't fit on the previous line, I like to line break before the `.`. So:

        someAsynchronousFunction(foo, bar, baz)
            .then(function (result) {
          doSomething(result);
        });